src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5; // range for random int values generator
+    private final Figure[] figures = new Figure[FIGURE_COUNT]; // array of possible figures
+    private final Random random = new Random(); // random int values generator
+
+    // constructor for array possible figures initialize
+    public FigureSupplier() {
+        // random figure color
+        ColorSupplier randomColor = new ColorSupplier();

```

Comment:
```suggestion
    private final Random random = new Random(); // random int values generator
    private ColorSupplier randomColor = new ColorSupplier();

    // constructor for array possible figures initialize
    public FigureSupplier() {
        // random figure color
        String randomColor = colorSupplier.getRandomColor();
```
ColorSupplier можна винести до рандому теж